examples/javascript/index.js

```diff
@@ -0,0 +1,140 @@
+var math = require('mathjs');
+
+data1 = [
+  [0, 10, 10, 80, 0],
+  [10, 0, 10, 0, 80],
+  [10, 10, 0, 40, 40],
+  [0, 0, 0, 1, 0],
+  [0, 0, 0, 0, 1],
+];
+
+data2 = [
+    [  1,   0,   0,   0,   0],
+    [0.5,   0, 0.5,   0,   0],
+    [  0, 0.5,   0, 0.5,   0],
+    [  0,   0, 0.5,   0, 0.5],
+    [  0,   0,   0,   0,   1],
+];
+
+function range(size, start_at= 0) {
+    return [...Array(size).keys()].map(i => i + start_at);
+}
+
+function decompose(matrix){
+    var qSize = numberOfTransients(matrix);
+    var rSize = matrix.size()[1] - qSize;
+    q = matrix.subset(math.index(range(qSize), range(qSize)));
+    r = matrix.subset(math.index(range(qSize), range(rSize, qSize)));
```

Comment:
I think you can accidentally overwrite global variables if you don't use **any** declaration keyword.